aws/resource_aws_security_group_rule.go

```diff
@@ -282,7 +298,7 @@ func resourceAwsSecurityGroupRuleRead(d *schema.ResourceData, meta interface{})
 	log.Printf("[DEBUG] Found rule for Security Group Rule (%s): %s", d.Id(), rule)
 
 	d.Set("type", ruleType)
-	if err := setFromIPPerm(d, sg, p); err != nil {
+	if err := setFromIPPerm(d, sg, rule); err != nil {
```

Comment:
This change is very subtle, but after reading the code up and down, I realized the current implementation actually would build an struct from the local state: `p` and then try and match all its parameters with the list of remotely fetched rules `rules` with function `findRuleMatch` returning `rule`. We then set the local state from `p`.... which is very strange because we build `p` from the local state... furthermore I feel its more "right" to set the local state from the remotely fetched data during the `Read` operation

Reply:
This change actually breaks the acceptance tests, I might take time to investigate which ones, if it becomes a mess I can confirm reverting the change fixes the tests, but as mentioned above something about this code feels very wrong.